sdk/src/main/java/com/akkaserverless/javasdk/eventsourcedentity/EventSourcedEntityEffect.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.eventsourcedentity;
+
+import com.akkaserverless.javasdk.Effect;
+import com.akkaserverless.javasdk.Metadata;
+import com.akkaserverless.javasdk.Reply;
+import com.akkaserverless.javasdk.ServiceCall;
+import com.akkaserverless.javasdk.impl.reply.FailureReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.ForwardReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.MessageReplyImpl;
+import com.akkaserverless.javasdk.impl.reply.NoReply;
+import com.akkaserverless.javasdk.reply.FailureReply;
+import com.akkaserverless.javasdk.reply.ForwardReply;
+import com.akkaserverless.javasdk.reply.MessageReply;
+
+import java.util.function.Function;
+
+/**
+ * A return type to allow returning forwards or failures, and attaching effects to messages.
+ *
+ * @param <T> The type of the message that must be returned by this call.
+ */
+public abstract class EventSourcedEntityEffect<T> implements Effect<T> {
+
+  public static class Builder<T, S> {
+    private Builder() {}
+
+    public Builder2<T, S> emitEvent(Object event) {
```

Comment:
`Object` here is not great, but there is no interface that all events would implement.

Reply:
A marker interface here?

Reply:
it's typically the generated proto messages that are the events

Reply:
Ah yes, thanks.